front 10.12

Oljk opened the pull request:
"1. в хедере делаем 3 кнопки: articles, add article, profile и 3 соответствующих компонента
2. по клику на одну из кнопок показываем соответствующий компонент и скрываем все остальные
*: на профиле пользователя делаем форму с 2мя полями (имя и фамилия) и после сабмита формы отображаем актуальное значение в хедере, на кнопке перехода в профиль «Profile (Ivan Ivanov)»"

Commit:
testing propagate functions to the child components

Commit:
add changing content via states

Commit:
adding user logic

mindk-pmishchenko commented on src/App.js:
```diff
@@ -1,27 +1,41 @@
     };
-//
+
+    const [page, setPage] = React.useState(views.articles);
```
лучше сразу деструктуризировать при импорте useState, чтоб не тагять за собой такую конструкцию

mindk-pmishchenko commented on src/components/content/article.jsx:
```diff
@@ -1,4 +1,5 @@
 
+
 function Article(props) {
     return (<div>{props.articleName} <br/> <ArticleContent articleText={props.articleText}/>  </div>);
```
лучше сразу деструктуризировать нужные пропсы

mindk-pmishchenko commented on src/components/header/contentUtils.js:
```diff
@@ -1,17 +1,32 @@
+import views from "../../views";
+import React from "react";
+
 function Logo(props) {
```
это должно быть в компонента, а не в utils - тут могут быть какие-то вычисления, обработки, но не компоненты

mindk-pmishchenko commented on src/components/header/user.js:
```diff
@@ -12,4 +16,21 @@ function HeaderAvatar(props) {
-export  {MiniUser, HeaderAvatar};
\ No newline at end of file
+
+function Profile(props) {
```
1 файл = 1 компонент

mindk-pmishchenko commented on src/components/header/user.js:
```diff
@@ -1,7 +1,11 @@
 return (<div>
-    <HeaderAvatar image={props.image} />  <a href={props.profileUrl}> {props.name} {props.surname} </a>
+    <HeaderAvatar image={props.image} />
+    <button onClick={() => {props.onclick(views.viewProfile)}}> {props.name} {props.surname}  </button>
```
договаривались избегать анонимных функций в рендере

mindk-pmishchenko replied:
с такой строкой могут быть проблемы. лучше выводить явно, тогда с ней ничего не случится
{` ${props.name} ${props.surname} `}

mindk-pmishchenko commented on src/containers/content/ContentGeneral.js:
```diff
@@ -1,8 +1,16 @@
+import views from "../../views";
+import {Profile} from "../../components/header/user";
 
 class ContentGeneral extends React.Component {
```
договаривались использовать функциональные компонеты

mindk-pmishchenko commented on src/containers/content/ContentGeneral.js:
```diff
@@ -1,8 +1,16 @@
 
 class ContentGeneral extends React.Component {
 
+/* грит юзлес конструктор
```
не оставляй ненужный закомментированный код

mindk-pmishchenko commented on src/containers/content/ContentGeneral.js:
```diff
@@ -16,10 +24,15 @@ class ContentGeneral extends React.Component {
-                    return (<Article articleName={article.articleName} articleText={article.articleText}/>);
-                })
+                {this.props.page === views.articles &&
+                this.articles.map((article, i) => {
```
плохая идея использовать счетчик в качестве key

mindk-pmishchenko replied:
старайся писать более простые конструкции 
`this.articles.map((article, i) => <Article articleName={article.articleName} articleText={article.articleText} key={i}/>)
`зачем там лишняя обертка?

mindk-pmishchenko commented on src/containers/header/ContentHeader.js:
```diff
@@ -10,6 +11,8 @@ function ContentHeader(props) {
  //   const name = {};
```
зачем это здесь?

Commit:
fixed:

https://github.com/Oljk/base_new_app_1/pull/3#discussion_r558844816
https://github.com/Oljk/base_new_app_1/pull/3#discussion_r558844248
https://github.com/Oljk/base_new_app_1/pull/3#discussion_r558844503

Commit:
разнесла все компоненты по отдельным файлам,
перевела анонимные функции в
 const click = () => {
        props.onclick(views.articles);
    }

fixed:
https://github.com/Oljk/base_new_app_1/pull/3#discussion_r558845212

https://github.com/Oljk/base_new_app_1/pull/3#discussion_r558845442
